odl/contrib/datasets/ct/mayo.py

```diff
@@ -47,43 +51,48 @@ def _read_projections(folder, indices):
 
     file_names = file_names[indices]
 
-    data_array = None
-
     for i, file_name in enumerate(tqdm.tqdm(file_names,
                                             'Loading projection data')):
         # read the file
-        dataset = dicom.read_file(folder + '/' + file_name)
+        try:
+            dataset = pydicom.read_file(folder + os.path.sep + file_name)
+        except:
+            print("corrupted file: {}".format(file_name), file=sys.stderr)
+            print("error:\n{}".format(sys.exc_info()[1]), file=sys.stderr)
+            print("skipping to next file..", file=sys.stderr)
+            continue
+
+        if not data_array:
+            # Get some required data
+            rows = dataset.NumberofDetectorRows
+            cols = dataset.NumberofDetectorColumns
+            rescale_intercept = dataset.RescaleIntercept
+            rescale_slope = dataset.RescaleSlope
+
+        else:
+            # Sanity checks
+            assert rows == dataset.NumberofDetectorRows
+            assert cols == dataset.NumberofDetectorColumns
+            assert rescale_intercept == dataset.RescaleIntercept
+            assert rescale_slope == dataset.RescaleSlope
 
-        # Get some required data
-        rows = dataset.NumberofDetectorRows
-        cols = dataset.NumberofDetectorColumns
-        hu_factor = dataset.HUCalibrationFactor
-        rescale_intercept = dataset.RescaleIntercept
-        rescale_slope = dataset.RescaleSlope
 
         # Load the array as bytes
         proj_array = np.array(np.frombuffer(dataset.PixelData, 'H'),
                               dtype='float32')
-        proj_array = proj_array.reshape([rows, cols], order='F').T
-
-        # Rescale array
-        proj_array *= rescale_slope
-        proj_array += rescale_intercept
-        proj_array /= hu_factor
-
-        # Store results
-        if data_array is None:
-            # We need to load the first dataset before we know the shape
-            data_array = np.empty((len(file_names), cols, rows),
-                                  dtype='float32')
-
-        data_array[i] = proj_array[:, ::-1]
+        proj_array = proj_array.reshape([cols, rows])
+        data_array.append(proj_array[:, ::-1])
         datasets.append(dataset)
 
+    data_array = np.array(data_array)
```

Comment:
Better use `np.stack`. If anything makes it impossible to build a properly-typed array, then `np.array` will give you one with `dtype=object` (which is horrible), whereas `np.stack` will fail (which is good).